### PR TITLE
Embed Binary Cache 

### PR DESCRIPTION
**Pull request by JehandadKhan**
Building upon #313 and #361 this PR embeds binary cache as well to serve compiled objects from within the lib file.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -115,6 +115,8 @@ set(MIOPEN_INSTALL_CXX_HEADERS Off CACHE BOOL "Install MIOpen's C++ header inter
 
 
 # Embedded Build Configuration
+set(MIOPEN_EMBED_DB "" CACHE STRING "Colon separated list of of architecture CU pairs to embed on-disk DBs in the binary. Example gfx906_60:gfx900_56")
```
Comment (daniellowell) [inline CMakeLists.txt:118]: PR #361 says it is semi-colon delimited. Which is it?

Comment (daniellowell) [reply CMakeLists.txt:118]: PING

Comment (JehandadKhan) [reply CMakeLists.txt:118]: Fixed

**Comment by daniellowell** [comment]
Tidy disaster!
http://micimaster.amd.com/blue/organizations/jenkins/MLLibs%2FMIOpen/detail/embedbincache/8/pipeline/

**Comment by daniellowell** [comment]
PING

Please update this PR as soon as you can.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -241,6 +241,34 @@ function(add_perf_test NAME)
 endfunction()
 
 set(IMPLICITGEMM_ARGS ${MIOPEN_TEST_FLOAT_ARG})
+# ./bin/MIOpenDriver conv -n 128 -c 1024 -H 14 -W 14 -k 2048 -y 1 -x 1 -p 0 -q 0 -u 2 -v 2 -l 1 -j 1 -m conv -g 1 -F 1 -t 1
+# MIOPEN_DEBUG_CONV_IMMED_FALLBACK=0
+if(MIOPEN_EMBED_DB)
+    set(MIOPEN_EMBED_TEST_ARG ${MIOPEN_TEST_FLOAT_ARG} --disable-validation --verbose)
```
Comment (atamazov) [inline test/CMakeLists.txt:247]: @JehandadKhan Could you please elaborate a bit why we need `--disable-validation` here?

Comment (JehandadKhan) [reply test/CMakeLists.txt:247]: we don't, this is an oversight.

Comment (atamazov) [reply test/CMakeLists.txt:247]: Please remove it if you have time or let me know and I will do so.

**Pull request merged**